### PR TITLE
Fix the javadoc doclint to fail on all errors

### PR DESCRIPTION
**Pull request by pieterbos**
Semaphore defaults to none, my mac to all, so let's fix it to all. 

confirmed, the current master now should be green with this PR applied, while this branch fails - since this PR actually adds the missing check to CI.

### CODE REVIEW

File: .semaphore/semaphore.yml [hunk 1]
```diff
@@ -11,19 +11,19 @@ blocks:
         - name: Java 8 build
           commands:
             - sem-version java 1.8
-            - ./gradlew build
+            - ./gradlew clean check javadoc build
```
Comment (J3173) [inline .semaphore/semaphore.yml:14]: I don't think clean is needed. The checkout should be clean anyway.

File: build.gradle [hunk 2]
```diff
@@ -26,6 +26,9 @@ allprojects {
   ext.gradleScriptDir = "${rootProject.projectDir}/gradle"
   //archivesBaseName = 'archie'
 
+  tasks.withType(Javadoc) {
+    options.addStringOption('Xdoclint:all', '-quiet')
+  }
```
Comment (J3173) [inline build.gradle:31]: Is this still needed?

Comment (pieterbos) [reply build.gradle:31]: Yes. On semaphore this defaults to -Xdoclint:none. The quiet should not be needed, but without it this does not work, so I added it again.

File: .semaphore/semaphore.yml (hunk 1 above)
Comment (pieterbos) [reply .semaphore/semaphore.yml:14]: I think this can go to juist build. i'll change it back.

Comment (pieterbos) [reply .semaphore/semaphore.yml]: removed the check javadoc (that's not necessary), kept the clean. Should be clean on checkout, but just to be sure.

**Review by J3173 — APPROVED** [review]

**Pull request merged**